### PR TITLE
[W4.3d][F11-A3]Lim Zeng Hou

### PR DESCRIPTION
**Pull request by zenghou**
- Split Address into more classes (i.e. Block, Street, Unit, PostalCode)
- Updated UserGuide.md to comply with address format
- Updated help command to comply with address format
- Updated expected.txt and input.txt to comply with address format

### CODE REVIEW

File: src/seedu/addressbook/data/person/SplitAddressByComma.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Splits address into an array containing Block, Street, Unit, Postal Code respectively
+ */
+
+public class SplitAddressByComma {
```
Comment (cmkumar87) [inline src/seedu/addressbook/data/person/SplitAddressByComma.java]: Why is this a class? This looks like a helper method. Helper methods are called within a class and are usually private methods. If the y have a larger Utility then make a Utility Class and add these as a static method

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -26,9 +31,18 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        initializeAddress(trimmedAddress);
     }
 
+    public void initializeAddress(String fullAddress) {
```
Comment (cmkumar87) [inline src/seedu/addressbook/data/person/Address.java]: again why is this public?

**Comment by cmkumar87** [comment]
please review and close PR.

File: src/seedu/addressbook/data/person/SplitAddressByComma.java (hunk 1 above)
Comment (zenghou) [reply src/seedu/addressbook/data/person/SplitAddressByComma.java]: I put this method in a class as I deemed it to have a greater utility. I had to split the address in each of the Block, Street, Unit, and Postal classes; copying the method as a helper method in each of these classes would be inefficient. However, as you mentioned, I should have made it a static method instead.

File: src/seedu/addressbook/data/person/Address.java (hunk 2 above)
Comment (zenghou) [reply src/seedu/addressbook/data/person/Address.java]: Ah yes, they should have been private instead.

**Comment by zenghou** [comment]
I have noted your comments, and will close this PR.

**Pull request closed without merging**